### ISSUE TITLE
hub.extraConfig.shell_config has no effect

### ISSUE THREAD

**Issue by stefanmeisner** [jupyterhub/zero-to-jupyterhub-k8s#2121]
### Bug description
I have seen many recommendations on how to configure the shell used in terminado, but none of them seems to be working (anymore?)
If I have a config.yml containing:
````
hub:
  extraConfig:
    shell_config:  |
      c.NotebookApp.terminado_settings = {'shell_command': ['/usr/bin/no_bash_here', '--login']}
````
I would expect troubles getting a terminal window. I have of course also tried more relevant examples. But whatever I write seems to be ignored without any error messages in the log.
I also tried to add the statement to jupyter_notebook_config.py in my Docker image, and that worked when I ran the image locally, but not when spawned from jupyterhub

#### Expected behaviour
That the snippet above would influense the behavior of terminado :)

#### Actual behaviour
Nothing happens

### How to reproduce
Insert the above
### Your personal set up
Running in AKS, helm chart 1.11, custom image
 - Dockerfile
FROM jupyterhub/k8s-singleuser-sample:0.11.1-n265.h9413a362

\# install dependencies
\# change user to execute script
USER root
RUN apt-get update && apt-get install -y gnupg gnupg2 gnupg1 --no-install-recommends
RUN wget https://packages.microsoft.com/keys/microsoft.asc -O key
RUN apt-key add key
RUN wget https://packages.microsoft.com/config/ubuntu/20.04/prod.list -O /etc/apt/sources.list.d/mssql-release.list
ENV ACCEPT_EULA=Y

RUN apt-get update && apt-get install -y \
    openssh-client \
    unzip \
    unixodbc \
    msodbcsql17 \
    mssql-tools \
    curl \
    texlive-xetex \
    && apt-get clean

\# Install Azure CLI
\# https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt
RUN /usr/bin/curl -sL https://aka.ms/InstallAzureCLIDeb | bash

RUN echo 'export PATH=$PATH:/opt/mssql-tools/bin' > /home/jovyan/.bash_profile
ENV JUPYTER_RUNTIME_DIR=/tmp

\# Switch back to jovyan to avoid accidental container runs as root
USER $NB_UID
RUN /opt/conda/bin/conda install --yes -c conda-forge panel
RUN /opt/conda/bin/conda install --yes -c conda-forge pandas
RUN /opt/conda/bin/conda install --yes -c conda-forge jupyter-server-proxy
RUN /opt/conda/bin/conda install --yes -c conda-forge jupyterlab-git
RUN /opt/conda/bin/conda install --yes -c conda-forge jupyter_conda
RUN /opt/conda/bin/conda install --yes -c conda-forge pyodbc
RUN /opt/conda/bin/conda install --yes -c conda-forge pyppeteer
RUN /opt/conda/bin/jupyter labextension install @jupyterlab/server-proxy
RUN /opt/conda/bin/jupyter lab build

 - OS:
 
 - Version(s):
 

- <details><summary>Full environment</summary>

```
# paste output of `pip freeze` or `conda list` here
```
</details>

- <details><summary>Configuration</summary>

```python
# jupyterhub_config.py
```
</details>

- <details><summary>Logs</summary>

```
# paste relevant logs here, if any
```
</details>

**Comment by consideRatio** [comment]
hub.extraConfig.<something> will be loaded by JupyterHub in a k8s pod named `hub-...`, but what you want to configure should be loaded by the server spawned by the JupyterHub in a separate container named `jupyter-<username>`.

This is possible to do in the latest development release of the Helm chart though.

Documentation about that is available here:
https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#singleuser-extrafiles

And it is available in for example chart version: `0.11.1-n349.he14d686f` (i just took the latest as listed here https://jupyterhub.github.io/helm-chart/#development-releases-jupyterhub)

Note that the documentation example is for adding a file into the hub pod, so it will be hub.extraFiles, but you want `singleuser.extraFiles`

```yaml
# I think this is the configuration you want, which works in version 0.11.1-n349.he14d686f and later
singleuser:
  extraFiles:
    jupyter_notebook_config:
      mountPath: jupyter_notebook_config.py
      stringData: |
        c.NotebookApp.terminado_settings = {'shell_command': ['/usr/bin/no_bash_here', '--login']}
```

**Comment by stefanmeisner** [comment]
Thanks alot for the help @consideRatio
